File: lua/p003.lua
function smallestFactor(x) 
	for i = 2, math.floor(math.pow(x, 0.5)), 1 do
		if x % i == 0 then
			return i
		end
	end
	return x  
end

function main() 
	x = 600851475143
	while true do
		y = smallestFactor(x)
		if y < x then
			x = x / y
		else 
			return x
		end
	end
end
print(main())

File: csharp/p003.cs
using System;
	
namespace csharp {

	class p003 {
		long smallestFactor(long x) {
			for (int i = 2; i <= Math.Floor(Math.Pow(x, 0.5)); i++) {
				if (x % i == 0) {
					return i;
				};
			};
			return x;  
		}
		public long run() {
			long x = 600851475143;
			while (true) {
				long y = smallestFactor(x);
				if (y < x) {
					x /= y;
				} else {
					return x;
				}
			}
		}
		static void Main(string[] args) {
			Console.WriteLine(new p003().run());	
		}
	}
}